communication/vehicleconnections/mavsdkstation.cpp

```diff
@@ -12,34 +13,41 @@ MavsdkStation::MavsdkStation(QObject *parent) : QObject(parent)
     mHeartbeatTimer.start(1000);
 
     mMavsdk.subscribe_on_new_system([this](){
+        static bool temporaryFixToStopOuterCallbackFromBeingCalledAgainBeforeTheInnerCallbackHasFinished = true;
         for (const auto &system : mMavsdk.systems()) {
-            if (!mVehicleConnectionMap.contains(system->get_system_id())) {
+            if (!mVehicleConnectionMap.contains(system->get_system_id()) && temporaryFixToStopOuterCallbackFromBeingCalledAgainBeforeTheInnerCallbackHasFinished) {
                 if (system->has_autopilot()) {
+                    temporaryFixToStopOuterCallbackFromBeingCalledAgainBeforeTheInnerCallbackHasFinished = false;
```

Comment:
I am not sure what the specific problem is here that makes this necessary. Will this callback for subscribe_on_new_system be called for all components of the system in parallel otherwise?
A better solution might be (but not sure I understand correctly) to let the subscribe_on_new_system send a Qt signal to a queued connection in order to serialize the calls and do the handling in that slot without the temporary fix.

Reply:
When having multiple systems discovered, subscribe_on_new_system will be called again, before the preceding system configuration has completed, and crash the application. To prevent this, the lock is necessary.